### PR TITLE
[sync] 2021/06/14

### PR DESCRIPTION
**Pull request by MeandNi**
ref #374

### CODE REVIEW

File: src/docs/codelabs/index.md [hunk 1]
```diff
@@ -1,10 +1,8 @@
 ---
-title: Codelabs
-title: Codelabs
-description: "Codelabs help you quickly get started programming Flutter."
-description: "Codelabs 是一种引导式的动手编码教程体验，帮助你更好的掌握 Flutter 编程技巧"
-tags: 教程, 代码实验室
-keywords: Flutter教程合辑,手把手教程,Flutter起步教程
+title: Codelabs & workshops
+title: Codelabs & workshops
```
Comment (AlexV525) [inline src/docs/codelabs/index.md:3]: 相同的是不是可以不写

File: src/docs/codelabs/index.md [hunk 2]
```diff
@@ -24,6 +33,14 @@ one of these codelabs:
 如果你刚开始学习 Flutter，
 我们推荐你学习下面的 Codelab 之一：
 
+* [Building your first Flutter app][]（workshop）**NEW!**<br>
+  An instructor-led version of our very popular
+  "Write your first Flutter app, part 1" codelab
+  (listed below).
+
+  [构建你的第一个 Flutter App][Building your first Flutter app] (workshop) **新!**<br>
+  我们非常受欢迎的「编写你的第一个 Flutter App，第一部分」 codelab （如下所列）的讲师指导版。
```
Comment (AlexV525) [inline src/docs/codelabs/index.md:42]:
```suggestion
  我们非常受欢迎的「编写你的第一个 Flutter App，第一部分」 codelab（如下所列）的讲师指导版。
```

File: src/docs/codelabs/index.md [hunk 3]
```diff
@@ -93,6 +111,42 @@ one of these codelabs:
...
+## Next steps
+
+## 下一步
+
+* [Building scrolling experiences in Flutter][] (workshop) **NEW!**<br>
+  Start with an app that performs simple, straightforward scrolling
+  and enhance it to create fancy and custom scrolling effects
+  by using slivers.
+
+  [在 Flutter 中构建滚动体验][Building scrolling experiences in Flutter]（workshop）**新!**<br>
```
Comment (AlexV525) [inline src/docs/codelabs/index.md:123]:
```suggestion
  [在 Flutter 中构建滚动体验][Building scrolling experiences in Flutter] (workshop) **新!**<br>
```

File: src/docs/codelabs/index.md [hunk 4]
```diff
@@ -93,6 +111,42 @@ one of these codelabs:
...
+  从一个执行简单、直接滚动的应用程序入门，
+  并通过使用 sliver 增强它以创建奇特的自定义滚动效果。
+
+* [Dart null safety in Action][] (workshop) **NEW!**<br>
+  An instructor-led workshop based on the [Null safety codelab][]
+  on the dart.dev site.
+
+  [Dart 空安全实战][Dart null safety in Action]（workshop）**新!**<br>
+  基于 dart.dev 站点上的 [空安全 codelab][] 的讲师指导版 workshop。
```
Comment (AlexV525) [inline src/docs/codelabs/index.md:132]: dart.cn ?

File: src/docs/codelabs/index.md [hunk 5]
```diff
@@ -93,6 +111,42 @@ one of these codelabs:
...
+
+  [在 Flutter 中构建滚动体验][Building scrolling experiences in Flutter]（workshop）**新!**<br>
+  从一个执行简单、直接滚动的应用程序入门，
+  并通过使用 sliver 增强它以创建奇特的自定义滚动效果。
+
+* [Dart null safety in Action][] (workshop) **NEW!**<br>
+  An instructor-led workshop based on the [Null safety codelab][]
+  on the dart.dev site.
+
+  [Dart 空安全实战][Dart null safety in Action]（workshop）**新!**<br>
```
Comment (AlexV525) [inline src/docs/codelabs/index.md:131]:
```suggestion
  [Dart 空安全实战][Dart null safety in Action] (workshop) **新!**<br>
```

File: src/docs/codelabs/index.md [hunk 6]
```diff
@@ -93,6 +111,42 @@ one of these codelabs:
...
+  基于 dart.dev 站点上的 [空安全 codelab][] 的讲师指导版 workshop。
+
+* [How to manage application states using inherited widgets][inherited-widget-ws] (workshop) **NEW!**<br>
+  Learn how to manage the state of your app's data by
+  using the `InheritedWidget` class, one of the
+  [low-level state management][] classes provided
+  by Flutter.
+
+  [如何使用 inherited widgets 管理应用程序状态][inherited-widget-ws]（workshop）**新!**<br>
```
Comment (AlexV525) [inline src/docs/codelabs/index.md:140]:
```suggestion
  [如何使用 inherited widgets 管理应用程序状态][inherited-widget-ws] (workshop) **新!**<br>
```

File: src/docs/codelabs/index.md [hunk 7]
```diff
@@ -93,6 +111,42 @@ one of these codelabs:
...
+  了解如何使用 `InheritedWidget` 类
+  （Flutter 提供的 [低级状态管理][low-level state management] 之一）
```
Comment (AlexV525) [inline src/docs/codelabs/index.md:142]: low-level 底层、基础

```suggestion
  （Flutter 提供的 [基础状态管理方案][low-level state management] 之一）
```

File: src/docs/codelabs/index.md [hunk 8]
```diff
@@ -235,13 +290,22 @@ Learn how to use Flutter with other technologies.
   Build a field trip app that allows you and other
   members of the trip to share photos.
 
-* [Build voice bots for mobile with Dialogflow and Flutter][] [NEW]<br>
+* [Build voice bots for mobile with Dialogflow and Flutter][dialogflow-ws] (workshop) **NEW!**<br>
```
Comment (AlexV525) [inline src/docs/codelabs/index.md:293]: 顺便翻了吧~

File: src/docs/cookbook/design/fonts.md [hunk 9]
```diff
@@ -173,7 +173,14 @@ outline weights and styles:
     the file as an integer multiple of 100, between 100 and 900.
     These values correspond to the [`FontWeight`][]
     and can be used in the [`fontWeight`][] property of a
-    [`TextStyle`][] object.
+    [`TextStyle`][] object. For example, if you want to use
```
Comment (AlexV525) [inline src/docs/cookbook/design/fonts.md:176]: 这里也顺便翻了吧

File: src/docs/deployment/cd.md [hunk 10]
```diff
@@ -85,6 +85,12 @@ Visit the [fastlane docs][fastlane] for more info.
    安装 fastlane `gem install fastlane` 或 `brew install fastlane`。
    访问 [fastlane docs][fastlane] 以获得更多信息。
 
+1. Create an environment variable named `FLUTTER_ROOT`,
+    and set it to the root directory of your Flutter SDK.
+    (This is required for the scripts that deploy for iOS.)
+
+    创建一个名为 `FLUTTER_ROOT` 的环境变量，并将其设置为 Flutter SDK 的根目录。（这是为 iOS 部署的脚本所必需的。）
```
Comment (AlexV525) [inline src/docs/deployment/cd.md:92]:
```suggestion
   创建一个名为 `FLUTTER_ROOT` 的环境变量，并将其设置为 Flutter SDK 的根目录。（这是为 iOS 部署的脚本所必需的。）
```

File: src/docs/development/data-and-backend/state-mgmt/options.md [hunk 11]
```diff
@@ -100,6 +100,17 @@ use under the hood.
 Widget tree 中不同层级间的 widget 通信的基础方法。这是诸如 `provider` 等
 众多方法的底层实现。
 
+The following instructor-led video workshop covers how to
+use `InheritedWidget`:
+
+以下讲师指导的视频 workshop 介绍了如何使用 `InheritedWidget`：
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/LFcGPS6cGrY" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (AlexV525) [inline src/docs/development/data-and-backend/state-mgmt/options.md:108]: cc @chenglu

File: src/docs/get-started/codelab.md [hunk 12]
```diff
@@ -34,6 +34,11 @@ diff2html: true
 
 {{site.alert.end}}
 
+If you prefer an instructor-led version of this codelab,
+check out the following workshop:
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/Z6KZ3cTGBWw" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (AlexV525) [inline src/docs/get-started/codelab.md:40]: cc @chenglu

File: src/index.html [hunk 13]
```diff
@@ -55,7 +55,9 @@ <h1 class="homepage__hero__text">Target <span class="color-white">mobile, web, &
 
                 <div class="homepage__intro__buttons">
                     <a class="btn btn-primary btn-cta" href="/docs/get-started/install">Get started</a>
+<!--
```
Comment (AlexV525) [inline src/index.html:58]: cc @chenglu

File: src/_assets/js/main.js [hunk 14]
```diff
@@ -1,4 +1,4 @@
-//= require vendor/jquery-3.3.1
+//= require vendor/jquery-3.6.0
```
Comment (AlexV525) [inline src/_assets/js/main.js:1]: 貌似指定 3.6.0 后 stable jekyll build 找不到这个版本

Comment (AlexV525) [reply src/_assets/js/main.js:1]: 需要更新 site-shared (submodules)

**Comment by chenglu** [comment]
看一下 [CI 的报错](https://github.com/cfug/flutter.cn/pull/895/checks?check_run_id=2814819408#step:12:21)：

```
YAML Exception reading /home/runner/work/flutter.cn/flutter.cn/src/posts/dart-2-7.md: (<unknown>): mapping values are not allowed in this context at line 2 column 19 
YAML Exception reading /home/runner/work/flutter.cn/flutter.cn/src/posts/announcing-dart-2-6.md: (<unknown>): mapping values are not allowed in this context at line 2 column 39
```

```
jekyll 3.9.0 | Error:  couldn't find file 'vendor/jquery-3.6.0' with type 'application/javascript' in `docs/release/breaking-changes/1-22-deprecations.md'
```

**Comment by AlexV525** [comment]
> 看一下 [CI 的报错](https://github.com/cfug/flutter.cn/pull/895/checks?check_run_id=2814819408#step:12:21)：
> 
> ```
> YAML Exception reading /home/runner/work/flutter.cn/flutter.cn/src/posts/dart-2-7.md: (<unknown>): mapping values are not allowed in this context at line 2 column 19 
> YAML Exception reading /home/runner/work/flutter.cn/flutter.cn/src/posts/announcing-dart-2-6.md: (<unknown>): mapping values are not allowed in this context at line 2 column 39
> ```
> 
> ```
> jekyll 3.9.0 | Error:  couldn't find file 'vendor/jquery-3.6.0' with type 'application/javascript' in `docs/release/breaking-changes/1-22-deprecations.md'
> ```

主站更改了 vendor 的集成方式，全部都丢到了 site-shared。以后每次 sync 最好都更新 submodules 了

**Comment by MeandNi** [comment]
> > 看一下 [CI 的报错](https://github.com/cfug/flutter.cn/pull/895/checks?check_run_id=2814819408#step:12:21)：
> > ```
> > YAML Exception reading /home/runner/work/flutter.cn/flutter.cn/src/posts/dart-2-7.md: (<unknown>): mapping values are not allowed in this context at line 2 column 19 
> > YAML Exception reading /home/runner/work/flutter.cn/flutter.cn/src/posts/announcing-dart-2-6.md: (<unknown>): mapping values are not allowed in this context at line 2 column 39
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > jekyll 3.9.0 | Error:  couldn't find file 'vendor/jquery-3.6.0' with type 'application/javascript' in `docs/release/breaking-changes/1-22-deprecations.md'
> > ```
> 
> 主站更改了 vendor 的集成方式，全部都丢到了 site-shared。以后每次 sync 最好都更新 submodules 了

sync ing.

File: src/docs/codelabs/index.md (hunk 1 above)
Comment (MeandNi) [reply src/docs/codelabs/index.md:3]: > 相同的是不是可以不写

之前也写了，这部分保留吧。

**Comment by MeandNi** [comment]
【两处 youtube 视频 cc @chenglu 】
src/docs/development/data-and-backend/state-mgmt/options.md
src/docs/get-started/codelab.md

【文章】
src/docs/development/ui/advanced/slivers
Using slivers to achieve fancy scrolling - 使用 sliver 实现出色的滑动效果
src/docs/null-safety.md
Null safety in Flutter - Flutter 中的空安全

**Review by AlexV525 — APPROVED** [review]
RSLGTM

File: src/docs/development/data-and-backend/state-mgmt/options.md (hunk 11 above)
Comment (chenglu) [inline src/docs/development/data-and-backend/state-mgmt/options.md:108]:
```suggestion
<iframe width="560" height="315" src="https://player.bilibili.com/player.html?aid=248744553&bvid=BV1Wv411W7yF&cid=354736130&page=1" title="Bilibili video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

File: src/docs/get-started/codelab.md (hunk 12 above)
Comment (chenglu) [inline src/docs/get-started/codelab.md:40]:
```suggestion
<iframe width="560" height="315" src="player.bilibili.com/player.html?aid=846249495&bvid=BV1n54y1H7dZ&cid=354773704&page=1" title="Bilibili video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

File: src/docs/null-safety.md [hunk 15]
```diff
@@ -34,9 +34,14 @@ class MyApp extends StatelessWidget {
 }
 ```
 
+For an instructor-led, free video workshop, check out the
+following:
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/HdKwuHQvArY" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (chenglu) [inline src/docs/null-safety.md:40]:
```suggestion
<iframe width="560" height="315" src="https://player.bilibili.com/player.html?aid=888693780&bvid=BV1tK4y1u76N&cid=354814166&page=1" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

File: src/docs/development/ui/advanced/slivers.md [hunk 16]
```diff
@@ -13,7 +13,14 @@ can define to behave in a special way.
 You can use slivers to achieve custom scrolling effects,
 such as elastic scrolling.
 
-For information on implementing fancy scrolling effects
+For a free, instructor-led video workshop that also uses DartPad,
+check out the following video about using slivers:
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/YY-_yrZdjGc" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (chenglu) [inline src/docs/development/ui/advanced/slivers.md:19]:
```suggestion
<iframe width="560" height="315" src="https://player.bilibili.com/player.html?aid=291195426&bvid=BV11f4y187gV&cid=354814353&page=1" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

File: src/docs/get-started/codelab.md [hunk 17]
```diff
@@ -34,6 +34,11 @@ diff2html: true
 
 {{site.alert.end}}
 
+If you prefer an instructor-led version of this codelab,
+check out the following workshop:
+
+<iframe width="560" height="315" src="player.bilibili.com/player.html?aid=846249495&bvid=BV1n54y1H7dZ&cid=354773704&page=1" title="Bilibili video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (chenglu) [inline src/docs/get-started/codelab.md:40]:
```suggestion
<iframe width="560" height="315" src="https://player.bilibili.com/player.html?aid=846249495&bvid=BV1n54y1H7dZ&cid=354773704&page=1" title="Bilibili video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

**Pull request merged**